stan/math/prim/functor/integrate_ode_rk45.hpp

```diff
@@ -21,12 +54,11 @@ inline auto integrate_ode_rk45(
     const std::vector<double>& x, const std::vector<int>& x_int,
     std::ostream* msgs = nullptr, double relative_tolerance = 1e-6,
     double absolute_tolerance = 1e-6, int max_num_steps = 1e6) {
-  internal::integrate_ode_std_vector_interface_adapter<F> f_adapted(f);
-  auto y = ode_rk45_tol_impl("integrate_ode_rk45", f_adapted, to_vector(y0), t0,
-                             ts, relative_tolerance, absolute_tolerance,
-                             max_num_steps, msgs, theta, x, x_int);
+  auto y = internal::integrate_ode_rk45_impl(f, y0, t0, ts, theta, x, x_int,
+                                             msgs, relative_tolerance,
+                                             absolute_tolerance, max_num_steps);
 
-  std::vector<std::vector<return_type_t<T_y0, T_param, T_t0, T_ts>>>
+  std::vector<std::vector<fn_return_type_t<F, T_y0, T_param, T_t0, T_ts>>>
```

Comment:
Yes. I did have _some_ logic for handling closures in `return_type_t` but for backwards compatibility some places need logic for also handling arbitrary C++ lambdas. I created `fn_return_type_t` because I wasn't sure if adding a catchall case for "anything that isn't a known type" would have some undesirable side effects. Your branch seems to have resolved that issue.